docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -1704,12 +1711,97 @@ unchanged, the package does not type check. Removing an interface from the
 version 2 package also causes issues, especially if the interface has
 choices.
 
-Instead, move interface and exception definitions out into a separate package
-from the start, such that subsequent versions of your package with templates all
-depend on the same version of the package with interfaces/exceptions. The SCU
-type checker warns about this, but you should see this warning as an error - it
-is very strongly recommended that you do not compile interfaces or exceptions in
-a package alongside templates.
+This means that template definitions that exist in the same package as
+interfaces and exception definitions will not be upgradeable. To avoid this
+issue, move interface and exception definitions out into a separate package in
+your model, such that subsequent versions of your package with templates all
+depend on the same version of the package with interfaces/exceptions.
+
+For example, a single package ``main`` defined as follows would not be able to
+upgrade, leaving the template ``T`` non-upgradeable.
+
+.. code:: daml
+
+  module Main where
+
+  interface I where
+    ...
+
+  template T with
+    ...
+
+.. code:: yaml
+
+  sdk-version: 2.10.0
+  name: main
+  version: 1.0.0
+  source: Main.daml
+  dependencies:
+  - daml-prim
+  - daml-stdlib
+  build-options:
+  - --target=1.dev
```

Comment:
I think I left this on the commit before, so leaving it here too.
This should be 1.17